src/seedu/addressbook/AddressBook.java

```diff
@@ -208,15 +208,35 @@
 
     public static void main(String[] args) {
         showWelcomeMessage();
-        processProgramArgs(args);
+        
+        // exercise 1
+        if (args.length >= 2) {
+            showToUser(MESSAGE_INVALID_PROGRAM_ARGS);
+            exitProgram();
+        }
+
+        if (args.length == 1) {
+            setupGivenFileForStorage(args[0]);
+        }
+
+        if(args.length == 0) {
+            setupDefaultFileForStorage();
+        }
         loadDataFromStorage();
-        while (true) {
+        
+        // exercise 2
+        userInputInstructions();
```

Comment:
This is arguable though. Making the loop into a method actually improves readability of the main :-).